### DISCUSSION TITLE
Connect a peripheral without scanning

### DISCUSSION THREAD

**Discussion by iyoruk31** [simpleble/simpleble#289] [Q&A] [answered]
Hi,

I would like to use simplepyble library. I will be grateful if you can provide an example of a script to connect a specific device without scanning. I could only find the same example everywhere that needs scanning of adapter and peripherals before connecting a device.

**Comment by eriklins** [comment, marked as answer]
From a BT/BLE specification perspective engaging a connection always requires prior scanning. A BLE advert is the invitation for engaging a connection and with scanning it won't be known to the central device when a connect window would be available. So, not possible without scanning.

**Reply by iyoruk31** [reply]
Thanks for the answer. Bleak API has a connection function without scanning (The scan might be done implicitly though). But it is pretty fast compare to the scan-and-connect. The problem about Bleak is that it uses async coroutines which clashes with Tkinter processes.  Simpleble works fine except connecting takes time as it involves scanning every time.
